reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java

```diff
@@ -88,16 +89,38 @@
 
 	@Override
 	public void doFirst() {
-		ContextSnapshot contextSnapshot = ContextSnapshot.capture(this.originalContext);
-
-		try (ContextSnapshot.Scope ignored = contextSnapshot.setThreadLocalValues()) {
-			this.scope = this.subscribeToTerminalObservation
-				.start()
-				.openScope();
-			//reacquire the scope from ThreadLocal
-			//tap context hasn't been initialized yet, so addToContext can now use the Scope
-			ContextSnapshot contextSnapshot2 = ContextSnapshot.capture(this.originalContext);
-			this.contextWithScope = contextSnapshot2.updateContext(Context.of(this.originalContext));
+		if (Micrometer.isContextPropagationAvailable()) {
```

Comment:
This might not always work very well. I think in some cases debuggers can introspect local variables which would cause classpath issues if context-propagation is not there. We tend to isolate such blocks of code in static, nested classes.

Reply:
can you elaborate? I'm not sure what and how to isolate at this level...